src/App.js

```diff
@@ -46,590 +53,453 @@ const languageProps = {
-          sanityTraining: JSON.parse(storedTrainingSanity),
-          sanityInterview: JSON.parse(storedInterviewSanity),
-          sanityProduct: JSON.parse(storedProductSanity),
-        });
-      }
-    } catch (e) {
-      console.log("Error fetching Sanity Experience: ", e);
-    }
-  };
+      let sprintStrings = [];
 
-  fetchStarterKitExperience = async (id) => {
-    try {
-      let experiences = await API.get("pareto", `/experience/user/${id}`);
-      let product;
-      let apprenticeship;
-      let interviewing;
-
-      experiences.forEach((exp) => {
-        if (exp.type === "Product") {
-          product = exp;
-        } else if (exp.type === "Apprenticeship") {
-          apprenticeship = exp;
-        } else if (exp.type === "Interviewing") {
-          interviewing = exp;
-        }
+      sprintData.map((spr, idx) => {
+        sprintStrings.push(`key${idx}=${spr.id}`);
       });
-      this.setState({
-        training: apprenticeship,
-        product: product,
-        interviewing: interviewing,
-        experiences: experiences,
-      });
-    } catch (e) {
-      console.log("Error fetching experience: ", e);
-    }
-  };
-
-  connectSocketToSprint = async () => {
-    const sprints = await API.get(
-      "pareto",
-      `/sprints/mentee/${this.state.user.id}`
-    );
-    this.props.getInitialSprintData(sprints);
-    this.setState({ sprints: sprints });
-
-    if (sprints.length === 0) {
-      return;
-    }
-
-    let sprintStrings = [];
-
-    sprints.map((spr, idx) => {
-      sprintStrings.push(`key${idx}=${spr.id}`);
-    });
 
-    let sprintString = sprintStrings.join("&");
+      let sprintString = sprintStrings.join("&");
 
-    let wsClient = new WebSocket(
-      `${process.env.REACT_APP_WSS_ENDPOINT}?${sprintString}`
-    );
+      const path = `${WSS_ENDPOINT}?${sprintString}`;
 
-    // console.log(wsClient);
+      const process = (message) => {
+        let tempSprintData = JSON.parse(message.data);
+        // this check is to see whether the websocket connection successfully retrieved the latest state.
+        // if there are too many extraneous connections, through ping error or otherwise - the function to distribute state across connections will fail
+        if (message && !tempSprintData.message) {
+          const index = sprintData.findIndex(
+            ({ id }) => id === tempSprintData.id
+          );
+          const newSprintArray = sprintData
+            .slice()
+            .splice(index, 1, tempSprintData);
+          try {
+            setSprints(newSprintArray);
+            putUpdatedSprintData(newSprintArray);
+          } catch (e) {
+            console.log("onmessage error", e);
+          }
+        } else {
+          // eslint-disable-next-line no-alert
+          alert(tempSprintData.message);
+        }
+      };
 
-    let that = this; // caching 'this'
-    let connectInterval;
+      ws.connect({ path, process });
+      ws.waitForConnection();
+    };
 
```

Comment:
Temporarily renamed initialFetch (and a couple other functions) to make it easier for me to keep a handle on what I was doing, but I revert them in a future commit.